pinot-common/src/main/java/org/apache/pinot/common/utils/LogicalTableUtils.java

```diff
@@ -74,7 +76,7 @@ public static ZNRecord toZNRecord(LogicalTableConfig logicalTableConfig)
   }
 
   public static void validateLogicalTableName(LogicalTableConfig logicalTableConfig, List<String> allPhysicalTables,
```

Comment:
Not introduced in this PR, but passing all physical table names as list and search in it is not efficient. Given this class is within the `pinot-common` package, we can pass in some lambda which returns a boolean indicating whether a table, tenant, schema exists.